recipes/hash-library/all/conanfile.py

```diff
@@ -13,17 +15,16 @@ class HashLibraryConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "compiler", "build_type", "arch"
     options = {
-        "shared": [True, False], 
+        "shared": [True, False],
         "fPIC": [True, False]
     }
     default_options = {
-        "shared": False, 
+        "shared": False,
         "fPIC": True
     }
-    generators = "cmake"
+    generators = "CMakeToolchain"
     exports_sources = ["CMakeLists.txt", "patches/*"]
 
-    _cmake = None
 
     @property
```

Comment:
please, remove both source_subfolder and build_subfolder, and use layout instead. Don't forget to update conandata.yml too, the patch there requires source_subfolder.

Reply:
These are baked into the build script 🙈